mangaki/mangaki/admin.py

```diff
@@ -229,60 +230,99 @@ def make_nsfw(self, request, queryset):
+            tag_weights = list(map(int, request.POST.getlist('weights')))
+            tag_anidb_tag_ids = list(map(int, request.POST.getlist('anidb_tag_ids')))
+            tags = list(map(AniDBTag, tag_titles, tag_weights, tag_anidb_tag_ids))
 
-            self.message_user(request, "Modifications sur les tags faites")
-            return None
+            # Checkboxes to know which tags have to be kept regardless of their pending status
+            tag_checkboxes = request.POST.getlist('tag_checkboxes')
+            tags_to_process = [tuple(map(int, tag_checkbox.split(':'))) for tag_checkbox in tag_checkboxes]
+
+            # Make a dict with work_id -> tags to keep
+            tags_final = {}
+            for index, work_id in enumerate(work_ids):
+                if work_id not in to_update_work_ids:
+                    continue
+                if work_id not in tags_final:
+                    tags_final[work_id] = []
+                if (work_id, tags[index].anidb_tag_id) in tags_to_process:
+                    tags_final[work_id].append(tags[index])
+
+            # Process selected tags for works that have been selected
+            for work in works:
+                if work.id in to_update_work_ids:
+                    client.update_tags(work, tags_final[work.id])
 
-        for anime in queryset.select_related("category"):
-            if anime.category.slug != 'anime':
+            if nb_updates == 0:
                 self.message_user(request,
-                                  "%s n'est pas un anime. La recherche des tags via AniDB n'est possible que pour les "
-                                  "animes " % anime.title)
-                self.message_user(request, "Vous avez un filtre à votre droite pour avoir les animes avec un anidb_aid")
-                return None
-            elif not anime.anidb_aid:
-                self.message_user(
-                    request,
-                    "%s n'a pas de lien actuel avec la base d'aniDB (pas d'anidb_aid)" % anime.title)
-                self.message_user(
-                    request,
-                    "Vous avez un filtre à votre droite pour avoir les animes avec un anidb_aid")
-                return None
+                                  "Aucune oeuvre n'a été marquée comme devant être mise à jour.",
+                                  level=messages.WARNING)
+            elif nb_updates == 1:
+                self.message_user(request,
+                                  "Mise à jour des tags effectuée pour une œuvre.")
+            else:
+                self.message_user(request,
+                                  "Mise à jour des tags effectuée pour {} œuvres.".format(nb_updates))
+            return None
+
+        # Check for works with missing AniDB AID
+        offending_works = []
+        if not all(work.anidb_aid for work in works):
+            offending_works = [work for work in works if not work.anidb_aid]
```

Comment:
You could rewrite this as `filter(lambda w: not w.anidb_aid, works)` and combine it with the `map` (which, if I recall well, are some kind of wrapper over generator expression which is memory efficient).

Simple alternative: just use generator expression `(work for … …)` and it'll be the same theorically.